### ISSUE TITLE
eapply causes Anomaly "Uncaught exception Not_found"

### ISSUE THREAD

**Issue by hsiang-wu** [coq/coq#9580]
#### Description of the problem

```
Require Import List.
Require Import Decidable PeanoNat.

Theorem count_occ_cons: forall l1 l2 (p1 p2:nat) n, count_occ Nat.eq_dec l1 p1 + n =
    count_occ Nat.eq_dec l2 p1 -> 
    count_occ Nat.eq_dec (p2 :: l1) p1 + n = count_occ Nat.eq_dec (p2 :: l2) p1.
Proof.
  intros. destruct (Nat.eq_dec p2 p1).
  - eapply eq_trans.
    + eapply eq_ind_r with (A:=nat) (x:= _) (P:= (fun x => x + n = _)).
      eapply eq_refl. eapply count_occ_cons_eq. apply e.
    + eapply eq_trans. (* <-- Error: Anomaly "Uncaught exception Not_found." *)
````
#### Coq Version
8.8.2 (Jan 2019)

**Comment by maximedenes** [comment]
Wow, this looks like it is a meta leak! Let's call @ppedrot to the rescue.

**Comment by Alizter** [comment]
Broken on 8.11, 8.12, 8.13. Fixed in 8.14. Therefore closing.
